From 8ebf40fe9bdc26f208789aa37454a2e940d94807 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jozef=20=C4=8C=C3=AD=C5=BE?= <jozef.ciz.ml@gmail.com>
Date: Sun, 12 Feb 2023 09:17:31 +0100
Subject: [PATCH] CHECK Change register behaviour registers

This behaviour suits me better. It's been too long since I changed this
so I am not sure what's the difference. Please consider what this does
and decide what to make of this change.
---
 src/mode.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/mode.cpp b/src/mode.cpp
index 26d96476..b1af43e6 100644
--- a/src/mode.cpp
+++ b/src/mode.cpp
@@ -113,8 +113,7 @@ void CommandContext::GetCommandRegisters()
     // No specified register, so use the default
     if (keymap.RegisterName() == 0)
     {
-        registers.push('*');
-        registers.push('+');
+        registers.push('"');
     }
     else
     {
@@ -1361,11 +1360,11 @@ bool ZepMode::GetCommand(CommandContext& context)
     }
     else if (mappedCommand == id_StandardPaste)
     {
+        context.pRegister = &GetEditor().GetRegister('+');
         if (context.currentMode == EditorMode::Visual)
         {
             context.replaceRangeMode = ReplaceRangeMode::Replace;
             context.op = CommandOperation::Replace;
-            context.pRegister = &GetEditor().GetRegister('"');
             auto range = GetInclusiveVisualRange();
             context.beginRange = range.first;
             context.endRange = range.second.Peek(1);
-- 
2.39.1

